### PR TITLE
linear burden test with new interface

### PR DESCRIPTION
**Pull request by jbloom22**
Now takes ``aggregate_expr`` of form ``gs.map(g => A).B``

### CODE REVIEW

File: src/main/scala/is/hail/methods/LinearRegressionBurden.scala [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      if (conflicts.nonEmpty)
+        fatal(s"Sample names conflict with these reserved statistical names: ${conflicts.mkString(", ")}")
+    }
+
+    val (variantKeysType, variantKeysQuerier) = vds.queryVA(variantKeySetVA)
+
+    if (variantKeysType != TSet(TString))
+      fatal(s"Variant keys must be of type Set(String), got $variantKeysType")
+
+    val pattern = """gs.map\(g => (.*)\).(.*)""".r
```
Comment (danking) [inline src/main/scala/is/hail/methods/LinearRegressionBurden.scala]: We are not parsing by regular expression.

The simplest thing is to just set `genotypeExpr` to `g` in the `makeKT` call.

Then, have the user write an expression that has `gs` in scope, call that `aggregateWith`, and write this:
```
val aggExpr = completeSamples.map(s => s"`$s` = let gs = `$s` in $aggregateWith").mkString(", ")
```

But actually don't like that either, it's way too easy for them to fuck up and include commas and get completely insane error messages. But I don't see a better option right now.

What we actually need is some way to parse the value side of a named expression independently of the name side.

Comment (danking) [inline src/main/scala/is/hail/methods/LinearRegressionBurden.scala]: And if you really don't want to put `g` in the `makeKT` call, just ask for a second expression that has `g` in scope and note that it occurs before aggregation, so pruning to just the necessary data will improve space usage.

File: src/main/scala/is/hail/methods/LinearRegressionBurden.scala [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          val qtx = QtBc.value * x
+          val qty = QtyBc.value
+          val xxp: Double = xx - (qtx dot qtx)
+          val xyp: Double = xy - (qtx dot qty)
+          val yyp: Double = yypBc.value
+
+          val b = xyp / xxp
+          val se = math.sqrt((yyp / xxp - b * b) / d)
+          val t = b / se
+          val p = 2 * T.cumulative(-math.abs(t), d, true, false)
```
Comment (danking) [inline src/main/scala/is/hail/methods/LinearRegressionBurden.scala]: I have not carefully examined this mathematics.

File: src/test/scala/is/hail/methods/LinearRegressionBurdenSuite.scala [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    2         .     2.0       .     2.0     0.0     0.0     2.0
+    3       0.0       .     1.0     1.0     1.0       .     3.0
+
+    Sample  Cov1    Cov2    Pheno   Gene1   Gene2   Gene3
+    A       0.0     -1.0    1.0     0.0     0.0     0.0
+    B       2.0     3.0     1.0     5.0     4.0     5.0
+    C       1.0     5.0     2.0     0.0     0.0     3.0
+    D       -2.0    0.0     2.0     4.0     4.0     7.0
+    E       -2.0    -4.0    2.0     0.0     0.0     3.0
+    F       4.0     3.0     2.0     1.0     0.0     1.0
```
Comment (danking) [inline src/test/scala/is/hail/methods/LinearRegressionBurdenSuite.scala]: This data is readily accessible from the files, including it here makes it easy to have the two sources go out of sync. I'm inclined to remove it.

**Comment by jbloom22** [comment]
Closing while I work on new version.

**Pull request closed without merging**